### PR TITLE
Adding 2.13 release notes

### PR DESCRIPTION
**Pull request by joshpalis**
### Issues Resolved
https://github.com/opensearch-project/flow-framework/issues/507

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-flow-framework.release-notes-2.13.0.0.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+## 2024-03-18 Version 2.13.0.0
```
Comment (amitgalitz) [inline release-notes/opensearch-flow-framework.release-notes-2.13.0.0.md:1]: I usually see this title in other release-notes: `## Version 2.13.0.0 Release Notes` is this also common to have date there?

Comment (joshpalis) [reply release-notes/opensearch-flow-framework.release-notes-2.13.0.0.md:1]: I took the same format as our 2.12 release notes

Comment (joshpalis) [reply release-notes/opensearch-flow-framework.release-notes-2.13.0.0.md:1]: Reference : https://github.com/opensearch-project/flow-framework/pull/467#discussion_r1471602226

Comment (amitgalitz) [reply release-notes/opensearch-flow-framework.release-notes-2.13.0.0.md:1]: I guess core does this but never seen a plugin have that, if it passed in 2.12 from build time then I am cool with it

**Review by amitgalitz — APPROVED** [review]

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**